### PR TITLE
API: Clients without queries should not be included

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Clients with zero queries should not be returned

---

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: src/api/stats.c [hunk 1]
```diff
@@ -391,7 +391,7 @@ int api_stats_top_clients(struct ftl_conn *api)
 		// Return this client if either
 		// - "withzero" option is set, and/or
 		// - the client made at least one query within the most recent 24 hours
-		if(includezeroclients || count > 0)
+		if(includezeroclients || client_count > 0)
```
Comment (DL6ER) [inline src/api/stats.c:394]: This is the actual bugfix in here. We were using the wrong variable. `count` is the maximum number of clients to return

**Review by yubiuser — DISMISSED** [review]
`withzero` is not documented

Screenshot at 2023-11-23 20-28-33

**Comment by DL6ER** [comment]
@yubiuser do we need it? It's there because it was there in the Telnet API but we don't know if anyone used it. I don't really see a need for this feature

**Comment by yubiuser** [comment]
Me neither. Then we should remove it. There should be no hidden/unused API parameters.

**Comment by rdwebdesign** [comment]
I noticed a small copy/paste error:

This comment:
https://github.com/pi-hole/FTL/blob/d5eb664e1f950ec22eeff20086b554105dc141be/src/api/stats.c#L232-L234

should be:
```diff
-// Check if this client should be skipped 
+// Check if this domain should be skipped 
```

**Comment by DL6ER** [comment]
@rdwebdesign Thanks, I even found another one.

**Review by rdwebdesign — APPROVED** [review]
Clients without queries are gone.

**Review by yubiuser — APPROVED** [review]

**Pull request merged**